### PR TITLE
Checkout section

### PR DESCRIPTION
**Pull request by harshvaghasiya-webelight**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -18,8 +19,15 @@ const App = () => {
         <Route path="about" element={<AboutSection />} />
         <Route path="products" element={<Products />} />
         <Route path="/products/:id" element={<SingleProduct />} />
-        <Route path="/cart" element={<ShoppingCart />} />
+        <Route
+          path="/cart"
+          element={<ShoppingCart activeState={0 as number} />}
```
Comment (anshs-webelight) [inline src/App.tsx:24]: why we need `as number`?

Comment (anshs-webelight) [reply src/App.tsx:24]: define enum for activeStates

File: src/App.tsx [hunk 2]
```diff
@@ -18,8 +19,15 @@ const App = () => {
...
+        />
         <Route path="/contact" element={<Contact />} />
+        <Route
+          path="/cart/checkout"
+          element={<CheckOut activeState={1 as number} />}
```
Comment (anshs-webelight) [inline src/App.tsx:29]: same as above

File: src/Context/PaymentContext.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import { createContext, useContext, useEffect, useMemo, useState } from "react";
+import { PaymentContextType, PaymentDetails } from "../Types/ProductsTypes";
+import { get, post } from "../utils/axios";
+import { v4 as uuid } from "uuid";
```
Comment (anshs-webelight) [inline src/Context/PaymentContext.tsx:4]: third party imports should be on top

File: src/Context/PaymentContext.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+const defaultState: PaymentContextType = {
+  paymentInfo: [],
+};
+
+const localStorageData = localStorage.getItem("PaymentState");
```
Comment (anshs-webelight) [inline src/Context/PaymentContext.tsx:10]: we should create a util file for all the localStorage operations

File: src/Context/PaymentContext.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [uState, setUState] = useState<PaymentContextType>(initialState);
+
+  const addPayment = async (pData: PaymentDetails) => {
+    const {
+      data: { key },
+    } = await get("http://localhost:4000/api/getkey");
```
Comment (anshs-webelight) [inline src/Context/PaymentContext.tsx:27]: we should create axios instance and add base URL there, so no need to add base URL every time

File: src/Context/PaymentContext.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    const { cartItems: _items, ...formData } = pData;
+
+    const payload = {
+      orderId: orderIds,
+      ...formData,
+    };
+
+    const {
+      data: { order },
+    } = await post("http://localhost:4000/api/checkout", payload);
```
Comment (anshs-webelight) [inline src/Context/PaymentContext.tsx:41]: same as above

File: package-lock.json
Comment (anshs-webelight) [inline package-lock.json:1]: package-lock.json is not needed we have yarn.lock already

File: src/main.tsx
Comment (anshs-webelight) [inline src/main.tsx:1]: file name should be index.tsx

File: src/modules/About/Components/AboutBanner.tsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import about_bg from "../../../images/about_bg.jpg";
```
Comment (anshs-webelight) [inline src/modules/About/Components/AboutBanner.tsx:1]: we should have index.ts file and export all the images from there

File: src/modules/Checkout/CheckOut.tsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+const CheckOut = ({ activeState }: { activeState: number }) => {
+  return (
+    <div>
+      <CartBanner activeState={activeState} />
+      <OrderDetails />
+    </div>
+  );
+};
+
+CheckOut.propTypes = {
```
Comment (anshs-webelight) [inline src/modules/Checkout/CheckOut.tsx:14]: In latest React development `.propTypes` is not needed and also you are using typescript already in your application

File: src/modules/Checkout/Components/OrderDetails.tsx [hunk 9]
```diff
@@ -0,0 +1,298 @@
+import { TextField } from "@mui/material";
+import { object, string, number, array } from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Controller, useForm } from "react-hook-form";
+import { CartItemType } from "../../../Types/ProductsTypes";
+import { useCartContext } from "../../../Context/CartContext";
+import { usePaymentContext } from "../../../Context/PaymentContext";
```
Comment (anshs-webelight) [inline src/modules/Checkout/Components/OrderDetails.tsx:7]: use absolute paths for import

File: src/Context/PaymentContext.tsx (hunk 5 above)
Comment (harshvaghasiya-webelight) [reply src/Context/PaymentContext.tsx:27]: i added base url but for payment service cuurently i am using diffrent server so i used diffrent url

**Pull request merged**